src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    private static final double DEFAULT_RADIUS = 10;
+    private final FigureType[] figureType = FigureType.values();
+    private final Random random = new Random();
+
+    public Figures getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        switch (figureType[random.nextInt(figureType.length)]) {
+            case CIRCLE:
+                return new Circle(random.nextDouble(),
+                        colorSupplier.getRandomColor());
+            case ISOSCALESTRAPECOID:
+                return new IsoscelesTrapezoid(random.nextDouble(),
+                        random.nextDouble(), random.nextDouble(),
+                        colorSupplier.getRandomColor());
+            case RECTANGLE:
+                return new Rectangle(random.nextDouble(), random.nextDouble(),
+                        colorSupplier.getRandomColor());
+            case RIGHTTRIANGLE:
+                return new RightTriangle(random.nextDouble(), random.nextDouble(),
+                        colorSupplier.getRandomColor());
+            default:
+                return new Square(random.nextDouble(),
+                        colorSupplier.getRandomColor());
+        }
+    }
+
+    public Figures getDefaultFigure() {
+        return new Circle(DEFAULT_RADIUS, DEFAULT_COLOR.toString());

```

Comment:
```suggestion
        return new Circle(DEFAULT_RADIUS, DEFAULT_COLOR.name());
```

Reply:
What is the difference between name() and toString()?

Reply:
name() for each enum value its own, toString() is common for each of them and if you override it, they'll have the same value, for example, you could write a text like this "default value" and for each of enum would be printed such value using toString()